azure/services/managedclusters/spec.go

```diff
@@ -305,6 +409,8 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existing interface{
 		}
 	}
 
+	managedCluster.AutoScalerProfile = buildAutoScalerProfile(s.AutoScalerProfile)
```

Comment:
@nojnhuh @mtougeron to speak to the thread about forward-compatibility and how to deal with clusters that have no declared `AutoScalerProfile` details (because prior versions of CAPZ didn't expose this) it might just be easier to do this:

```go
if s.AutoScalerProfile != nil {
    managedCluster.AutoScalerProfile = buildAutoScalerProfile(s.AutoScalerProfile)
}

```

Reply:
At first glance that seems equivalent to the condition currently inside `buildAutoScalerProfile` which comes with the same limitation that once any value is set, none of them can ever be reset to back to AKS's implicit defaults:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/47d8731b86cb220ee4bfee7d833b4a7a90e7abf9/azure/services/managedclusters/spec.go#L227-L229

Reply:
Ah, sorry, I thought we were in the flow of reconciling existing AKS API data w/ local CAPZ resource data.

So, what I'm thinking is: can we passively apply AKS API's version of the state if the CAPZ resource data is empty, and only definitively update the AKS API if we actually have an opinion?

Reply:
I think the nuance here is that a possible opinion is "reset to the implicit defaults" which is indistinguishable from "use what's already defined in AKS" in the CAPZ API. Is there a way to construct the YAML such that the controller ultimately reads a non-nil, empty value from the API? If something like
```yaml
autoscalerProfile: {}
```
is non-nil and omitting it entirely (or `null`) is nil, then I think everything can work the way we want as-is. I vaguely remember trying something similar a long time ago and `{}` still came through as nil.

Reply:
I tried setting an explicit empty map like `autoscalerProfile: {}` in the YAML and that actually comes through as a non-nil, empty struct value the way I hoped. The defaulting in the webhook still populates all the fields in that case though, so that can't be used to reset all the fields back to the implicit defaults. And the defaulting is necessary to accurately detect when we need to trigger an update to the AKS instance.

I suppose there are workarounds we could experiment with like treating `{}` as a special value in the webhook or moving CAPZ's awareness of the defaults from the webhook further down closer to here, but those are hacky enough that they're not worth adding on a first pass and could maybe be introduced in a backwards-compatible manner later.

Sorry @mtougeron if I'm only now understanding things you figured out a while ago, but I think I at least have a more solid grasp on things now.

Reply:
I think the tl;dr here for me is let's punt this out of this PR's scope for now, I think this is (unfortunately) one of many instances of this "property of properties" challenge that we have wrapping AKS config. Aspirationally we can come up with an approach that generalizes across all of these for CAPZ + AKS?